src/main/java/configurationslicing/docker/AbstractDockerSlicerSpec.java

```diff
@@ -0,0 +1,185 @@
+    public abstract String getSliceParam(DockerBuilder builder);
+    public abstract DockerBuilder setSliceParam(DockerBuilder builder, String value);
+
+    @Override
+    public String getDefaultValueString() {
+        return DEFAULT;
+    }
+
+    @Override
+    public String getName(AbstractProject<?, ?> item) {
+        return item.getDisplayName();
+    }
+
+    public List<AbstractProject<?, ?>> getWorkDomain() {
+        List<AbstractProject<?, ?>> list = new ArrayList<AbstractProject<?, ?>>();
+        List<AbstractProject> temp = Jenkins.getInstance().getAllItems(AbstractProject.class);
+        for (AbstractProject p: temp) {
+            if (p instanceof Project || p instanceof MatrixProject) {
+                list.add(p);
+            }
+        }
+        return list;
+    }
+
+
+    @Override
+    public List<String> getValues(AbstractProject<?,?> item) {
+
+        List<String> valuesList = new ArrayList<String>();
+
+        DescribableList<Builder, Descriptor<Builder>> buildersList = getBuildersList(item);
+        List<DockerBuilder> builders = getDockerBuildersList(buildersList);
+
+        for (DockerBuilder builder: builders) {
+            valuesList.add(getSliceParam(builder));
+        }
+        if (valuesList.isEmpty()) {
+            valuesList.add(NOTHING);
+        }
+
+        return valuesList;
+    }
+
+    @Override
+    public boolean setValues(AbstractProject<?,?> item, List<String> values) {
+
+        DescribableList<Builder, Descriptor<Builder>> buildersList = getBuildersList(item);
+        List<DockerBuilder> dockerBuildersList = getDockerBuildersList(buildersList);
+
+        int maxListSize = Math.max(values.size(), dockerBuildersList.size());
+        DockerBuilder[] oldBuilders = new DockerBuilder[maxListSize];
+        DockerBuilder[] newBuilders = new DockerBuilder[maxListSize];
+
+        for (int i = 0; i < dockerBuildersList.size(); i++) {
+            oldBuilders[i] = dockerBuildersList.get(i);
+        }
+
+        for (int i = 0; i < values.size(); i++) {
+            String value = values.get(i);
+            newBuilders[i] = oldBuilders[i];
+
+            if(oldBuilders[i] != null && !getSliceParam(oldBuilders[i]).equals(value)) {
+                if(valueIsNotWritable(value)) {
+                    value = "";
+                }
+                newBuilders[i] = setSliceParam(newBuilders[i], value);
+            }
+        }
+
+        // perform any replacements
+        for (int i = 0; i < maxListSize; i++) {
+            if (oldBuilders[i] != null && newBuilders[i] != null && oldBuilders[i] != newBuilders[i]) {
+                replaceBuilder(buildersList, oldBuilders[i], newBuilders[i]);
+                log.info(
+                        "updated DockerBuilder config {} in project {}",
+                        getUrl(),
+                        item.getFullName()
+                );
+            }
+        }
+
+        // add any new ones (can't see when this would happen)
```

Comment:
I didn't find anything that could display an error as part of the page so you'd know instantly but it would be in the logs.

I've decided to remove the delete builder part because i do newBuilders[i] = oldBuilders[i] and theres no situation which would set newBuilder[i] to null which makes the delete check obsolete. The adding builder error message is now more verbose.